specification/cognitiveservices/data-plane/EntitySearch/v1.0/EntitySearch.json

```diff
@@ -0,0 +1,935 @@
+            "required": false,
+            "type": "string",
+            "enum": [
+              "Off",
+              "Moderate",
+              "Strict"
+            ],
+            "x-ms-enum": {
+              "name": "SafeSearch",
+              "modelAsString": false
+            }
+          },
+          {
+            "name": "SetLang",
+            "in": "query",
+            "description": "The language to use for user interface strings. Specify the language using the ISO 639-1 2-letter language code. For example, the language code for English is EN. The default is EN (English). Although optional, you should always specify the language. Typically, you set setLang to the same language specified by mkt unless the user wants the user interface strings displayed in a different language. This parameter and the Accept-Language header are mutually exclusive; do not specify both. A user interface string is a string that's used as a label in a user interface. There are few user interface strings in the JSON response objects. Also, any links to Bing.com properties in the response objects apply the specified language.",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/SearchResponse"
+            }
+          },
+          "400": {
+            "description": "Invalid request.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "401": {
+            "description": "Invalid Authorization.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "403": {
+            "description": "Insufficient Authorization.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "429": {
+            "description": "Rate Limit Exceeded.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "500": {
+            "description": "Server Error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SearchResponse": {
```

Comment:
Recommendation: less nesting

This definition is more concise and carries more semantical information if you model it as
```
"description": "Defines the top-level object that the response includes when the request succeeds.",
"type": "object",
"allOf": [
  {
    "$ref": "#/definitions/Response"
  }
],
"properties": {
  "queryContext": {
    "description": "...",
    "$ref": "#/definitions/QueryContext"
  },
  "entities": {
    "description": "A list of entities that are relevant to the search query.",
    "$ref": "#/definitions/Entities"
  },
  "places": {
    "description": "A list of local entities such as restaurants or hotels that are relevant to the query.",
    "$ref": "#/definitions/Places"
  }
}
```

While generated code is the same (at least for AutoRest), this is smaller, has less curlies and stuff, but most importantly more clearly expresses the inheritance relationship:
> `SearchResponse` inherits from `Response` and has the following properties ...

Reply:
If a type is used as a response only, please model all of it's properties as `readOnly: true`.

Reply:
Re: less nesting, the definitions are generated dynamically out of our schema files that live in-branch. We definitely don't want to artificially flatten structures because this code generation will be re-used across the multiple APIs we offer, each with a different superset of models that could potentially rely on intermediately inherited types. Artificially flattening will definitely lead to issues. Regarding the readOnly, that will also be interesting, since we want to auto generate most of the stuff, but it's doable :)

Reply:
Oh I misunderstood, you mean to switch to filling in the object's .properties instead of putting the object's properties inside a new object (with new curlies) inside it's .allOf? I noticed some people do it that way (and it does indeed make WAY more sense), but it goes against the swagger specification's guidelines https://swagger.io/specification/. I built our spec generator to follow what's on the swagger site :|

Reply:
> but it goes against the swagger specification's guidelines

Oh! I (and probably more people) were not aware of that, 99% of this repo go the less nesting route 😮 Where in the spec do you see that guideline?

Reply:
I dunno if "guideline" is the right word, it's just how they show all of their inheritance / polymorphism examples. (just search for 'allOf' in the swagger.io/specification)

Reply:
Fair. I discussed with the others, please reduce the nesting for readability and consistency with the remaining repo.